plugins/include/engine.inc

```diff
@@ -24,242 +24,1271 @@
+ *
+ * @note For a list of available flags see the SPEAK_* constants in
+ *       engine_const.inc
+ *
+ * @param iIndex        Client index
+ * @param iSpeakFlags   New flags to set
+ *
+ * @noreturn
+ * @error               If the client index is not within the range of 1 to
+ *                      MaxClients, or the client is not connected, an error
+ *                      will be thrown.
+ */
 native set_speak(iIndex, iSpeakFlags);
+
+/**
+ * Returns the engine module speak flags currently set on a client.
+ *
+ * @note For a list of available flags see the SPEAK_* constants in
+ *       engine_const.inc
+ *
+ * @param iIndex    Client index
+ *
+ * @return          Client speak flags
+ * @error           If the client index is not within the range of 1 to
+ *                  MaxClients, or the client is not connected, an error will be
+ *                  thrown.
+ */
 native get_speak(iIndex);
 
-/* Drops an entity to the floor (work?) */
+/**
+ * Uses the DROP_TO_FLOOR() engine function on an entity, which attempts to put
+ * it down on the floor.
+ *
+ * @note This engine function traces 256 units straight downwards from the
+ *       entity origin. If the trace hits the floor, the origin is updated to
+ *       the end position of the trace, FL_ONGROUND is added to the flags and
+ *       EV_ENT_groundentity is updated. When the trace does not hit anything or
+ *       the entity would be stuck inside something, the process fails.
+ *
+ * @param entity    Entity index
+ *
+ * @return          1 if entity is on the floor, 0 otherwise
+ */
 native drop_to_floor(entity);
 
-/* Get whole buffer containing keys and their data. */
+/**
+ * Retrieves keyvalue buffer from a client or the server.
+ *
+ * @note There are three different types of keyvalue buffers, depending on the
+ *       index passed:
+ *           -1 - "local" buffer (various server information and config values)
+ *            0 - server buffer (usually contains "*gamedir" only)
+ *           >0 - client buffer ("name", "rate" and other client info)
+ * @note The buffer is formatted as "\key1\value1\key2\value2\...\keyN\valueN"
+ *
+ * @param id        Server/client index
+ * @param buffer    Buffer to copy keybuffer to
+ * @param length    Maximum size of buffer
+ *
+ * @return          Number of cells written to buffer
+ * @error           If an invalid entity index is provided or, if the index is a
+ *                  client index, the client is not connected, an error will be
+ *                  thrown.
+ */
 native get_info_keybuffer(id, buffer[], length);
 
-/* Use an entity with another entity. "entUsed" could be a hostage, "entUser" a player. */
+/**
+ * Forces an entity (such as a player) to use another entity (such as a button).
+ *
+ * @param entUsed   Index of entity being used
+ * @param entUser   Index of entity using
+ *
+ * @noreturn
+ * @error           If an invalid entity index is provided or, if the index is a
```

Comment:
I meant it's confusing which index you're talking about. You should specify if it's entUsed or entUser.